### DISCUSSION TITLE
Desenvolvendo um front-end para 224 países e mais de 40 línguas

### DISCUSSION THREAD

**Discussion by leonardoelias** [frontendbr/forum#322] [Dicas]
Galera, esse ano fui ao E-commerce Brasil, com o pessoal da _FIRMA_, e me surpreendi com a  palestra do Eduardo Shiota que foi ao ar no youtube, _segue o link_: https://www.youtube.com/watch?v=MVkRJgPxLjM

**Comment by davidsonfellipe** [comment]
@leonardoelias O @eshiota mandou muito, além de ser a paciência em pessoa.

**Comment by eshiota** [comment]
Valeu gente ❤️ 

Se vocês tiverem quaisquer dúvidas sobre o que eu falei, podem mandar que eu vou tentar responder. Tem certas coisas que não podemos compartilhar, mas vou fazer o máximo pra sanar todas as perguntas 🙌

**Comment by felipefialho** [comment]
Ta fácil entra as palestras mais legais que já vi. Assisti na WebBR 2014, e tinha um MONTE de dicas super legais, um monte de informação e conceitos valiosos e um monte de exemplos culturais.

Simplesmente sensacional.

**Comment by robsonbittencourt** [comment]
Muito boa a palestra @eshiota.

Uma dúvida que eu fiquei: essa grande quantidade de coisas que vocês fazem track, não acabam gerando um overhead para o usuário ou até para o sistema em si?

**Comment by eshiota** [comment]
@LFeh Essa palestra do Fórum E-commerce tem um título similar, mas o conteúdo é bem diferente (falo um pouco de localização, mas é mais focado na cultura de front-end na Booking e como fazemos com que tudo funcione na escala da nossa operação)

@robsonbittencourt No server, a arquitetura que suporta o tracking permite que façamos isso sem impacto aos usuários; no client os trackings são assíncronos e não bloqueiam nada na experiência de navegação.

**Comment by fititnt** [comment]
@eshiota você falou na palestra do Canadá. Mas não estou vendo bandeira do Canadá nos idiomas.

OMG, minha vida é um teste A/B. Isso parece matrix

![captura de tela de 2016-10-11 13-19-13](https://cloud.githubusercontent.com/assets/812299/19278952/b5f890fc-8fb5-11e6-9cc6-12b73f3fa43e.png)

**Comment by eshiota** [comment]
@fititnt Você está no Canadá (pergunta não troll, de verdade 😃 )? Essa é a condição pra ver a bandeira do Canadá nas opções English (UK), English (US) e French. Todos os visitantes de outros países vão ver as bandeiras dos EUA e Reino Unido.

**Comment by fititnt** [comment]
Não, não estou. E faz total sentido, porque se tivesse opções demais, eu ficaria confuso. Já tem muita opção na janela. Eu gostei também que vocês deixam uma lista das mais acessadas.

---

Outro ponto. Pra quem tiver interesse, ele fala disso em 7min55s https://youtu.be/MVkRJgPxLjM?t=475

Na sua palestra você comentou rapidamente, mais exatamente dedicou 30 segundos, mas eu tenho certeza que a equipe de vocês tomou cuidado é com relação ao uso de **bandeira de país para representar uma língua**.

Tem toda uma polêmica no campo da internacionalização de que bandeiras não são ideais pra representar línguas, por questões que você falou ali de que "um país pode ter mais de uma língua".

Embora você tenha falado pouco, por questões que são perceptíveis, pois estava em um evento de E-commerce, não muito focado em internacionalização, vi que no site mesmo vocês tiveram uma abordagem que me parece bem bacana: 
- **Usam texto para descrever qual é o idioma** (são explícitos, isso vai de encontro ao que outros grupos de i18n falam)
- **Usam bandeira _também_** (consigo imaginar que isso ajuda identificar de forma rápida, eu me identifiquei com bandeira do Brasil de imediato, me causou empatia)

Caso você no futuro for dar outra palestra desse tipo que o foco seja internacionalização, pode falar um pouco mais explicitamente das formas de **representação de idioma** e **de moeda**? Digo, de explicitamente falar de porque apenas bandeira é ruim, mas ao mesmo tempo vocês _ainda assim_ usam bandeira?

Referências pra ter uma ideia
- http://i18n-html-tech-lang.pt.webiwg.org/#linkdestination
- https://www.w3.org/TR/i18n-html-tech-lang/#linkdestination
- https://www.w3.org/International/techniques/authoring-html#linkdestination
- http://bittersmann.de/articles/no-flags/

**Comment by eshiota** [comment]
@fititnt Temos a vantagem de ter todo um sistema de testes A/B por trás das decisões, e também coletamos dados qualitativos por meio de user testing e outras técnicas. Portanto, nossa decisão não é baseada somente em artigos (claro, é sempre bom ter o conhecimento de diferentes especialistas), e sim no que o usuário realmente precisa. Analisando métricas como: número de vezes que o usuário trocou de idioma, bounce rate, e net conversion podemos ter a certeza do impacto positivo ou negativo das mudanças de design que fazemos.

Tudo o que está no nosso site é basicamente o resultado de decisões baseadas em dados e feedback dos usuários.

**Comment by fititnt** [comment]
Sim. Não é de hoje que sei que A/B é forte da Booking. Por isso que creio que dá pra ir um pouco mais além e usar experiência de vocês e discutir outras abordagens de um tema muito específico, relacionado a como indicar idiomas ou moedas de forma que fique claro.

O [artigo de 2011 About languages and flags](http://bittersmann.de/articles/no-flags/) que é usado como referência de boas práticas talvez não gere tanto impacto a ponto de valer a pena ser usado exatamente como se esperava 5 anos atrás.

Além do Brasil, existem outras conferências sobre padrões de código, em especial nos EUA e na Europa. Vocês na Booking tem um caso de uso bem interessante. Se você topar estudar um pouco sobre isso, como com ajuda de um grupo de colegas meus, e topar documentar métricas ou discussões dentro da sua empresa de decisões de porque fizeram de um jeito ou de outro, eu tenho quase certeza que eu te garanto o seguinte:
- Depois de revisão de outras pessoas, parte das conclusões da Booking vão se tornar boas práticas da W3C
- Você ser convidado a dar palestra sobre o tema internacionalização em 2017 fora do Brasil

**Comment by RuanAragao** [comment]
Olá @eshiota, primeiramente parabéns e obrigado, pois já absorvi bastante informação com você. Estou tendo minha primeira real experiencia com o mundo do e-commerce, gostaria de saber se atualmente vocês utilizam alguma ferramenta de terceiros para auxiliar na coleta de informações, analise e testes A/B, como por exemplo Hotjar, CrazyEgg, Optimizely e afins.

**Comment by saitodisse** [comment]
Bela palestra @eshiota, parabéns! @RuanAragao eu acho que eles utilizam Elastic Search com Kibana, mas é um chute, hehehe. Mas tbm tenho essa dúvida em relação a como pegar os dados do cliente no browser.

Opa, achei um blog post sobre ferramentas que o pessoal da Booking usa: http://blog.booking.com/ok-monitoring-enabled.html

**Comment by santosluiz** [comment]
Assunto interessantíssimo, obrigado por compartilhar! Salvo para assistir mais tarde ;)
